### PR TITLE
save all the models

### PR DESCRIPTION
**Pull request by msultan**
- [x] Implement feature / fix bug
 - [x] Add tests
 - [x] Update changelog

Adds option to bootstrap msm to optionally save all models.

### CODE REVIEW

**Comment by mpharrigan** [comment]
Why does this need to be configured? Can't we just do it? Can we spell out "model"

Other things I noticed in this class:

 -  fit should return `self`
 - the msm kwargs shouldn't have a default param of `{}` because python is dumb and then this dictionary becomes suprisingly mutable. Do the normal thing of setting it to `None` and then checking `if msm_kwargs is None: msm_kwargs = {}`
 - There's a superfluous return https://github.com/msultan/msmbuilder/blob/0d1d9b35a3a4402f6d1e683a0c005a716d8f55b6/msmbuilder/msm/validation/bootstrapmsm.py#L164

**Comment by mpharrigan** [comment]
Usually when we add new arguments, we put them at the end for backwards compatibility

**Comment by msultan** [comment]
> Why does this need to be configured? 

Because, it might increase the object size by a lot. 

>Can't we just do it? Can we spell out "model"

Fine, but mdls is the web3.0/start-up version of this . 

> Other things I noticed in this class:
> 
> fit should return self

kk

> the msm kwargs shouldn't have a default param of {} because python is dumb and then this dictionary becomes suprisingly mutable. Do the normal thing of setting it to None and then checking if msm_kwargs is None: msm_kwargs = {}

Didnt know this,changing it now. 

> There's a superfluous return https://github.com/msultan/msmbuilder/blob/0d1d9b35a3a4402f6d1e683a0c005a716d8f55b6/msmbuilder/msm/validation/bootstrapmsm.py#L164

**Comment by msultan** [comment]
No idea why py3.5 is failing and 3.4 is working somehow.

File: docs/changelog.rst [hunk 1]
```diff
@@ -6,6 +6,11 @@ Changelog
 v3.7 (development)
 ------------------
 
+New Features
+~~~~~~~~~~~~
+
+-``BootStrapMarkovStateModel`` can now optionally save all the
```
Comment (cxhernandez) [inline docs/changelog.rst]: +1

File: msmbuilder/msm/validation/bootstrapmsm.py [hunk 2]
```diff
@@ -107,7 +114,7 @@ def fit(self, sequences, y=None, pool=None):
         sequences = list_of_1d(sequences)
         self.mle_.fit(sequences, y=y)
         self._parallel_fit(sequences, pool)
-
+        return self
```
Comment (cxhernandez) [inline msmbuilder/msm/validation/bootstrapmsm.py]: ?

**Comment by mpharrigan** [comment]
How big could all the models possibly be? The way it's set up now, they have to all fit in RAM simultaneously. A couple more megabytes on disk isn't going to hurt anyone (how much space are we talking here?). What about encouraging users to 
```python
msm.fit()
msm.all_models_ = None
dump(msm)
```
where space is a premium?

**Comment by mpharrigan** [comment]
Should this be `msm.all_models_` (instead of `msm.all_models`)?

**Comment by msultan** [comment]
for a 300 stat-space, its about 100MB for 50 bootstrap samples.

**Comment by msultan** [comment]
I think if users run 1000s of bootstrap samples, this might get large.

**Comment by msultan** [comment]
i changed the behavior the mapped populations so that it defaults to np.nan instead of 0s.

**Comment by msultan** [comment]
@msmbuilder/developers anyone have any ideas as to why the featurizer tests are failing on p3.5 and 2.7?

**Comment by mpharrigan** [comment]
it's sklearn 0.18  see #918

**Comment by mpharrigan** [comment]
Is this mergeable?

**Comment by msultan** [comment]
yea, tests are still failing though.

**Comment by mpharrigan** [comment]
They aren't relevant to this PR

**Pull request merged**